Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -243,19 +238,16 @@ class SerializerTest : public ::testing::Test {
   }
 
   ByteString EncodeFieldValue(const FieldValue& fv) {
-    ByteStringWriter writer;
-    google_firestore_v1_Value proto = serializer.EncodeFieldValue(fv);
-    writer.Write(google_firestore_v1_Value_fields, &proto);
-    FreeNanopbMessage(google_firestore_v1_Value_fields, &proto);
-    return writer.Release();
+    return Encode(serializer.EncodeFieldValue(fv));
   }
 
   ByteString EncodeDocument(const DocumentKey& key, const ObjectValue& value) {
-    ByteStringWriter writer;
-    google_firestore_v1_Document proto = serializer.EncodeDocument(key, value);
-    writer.Write(google_firestore_v1_Document_fields, &proto);
-    FreeNanopbMessage(google_firestore_v1_Document_fields, &proto);
-    return writer.Release();
+    return Encode(serializer.EncodeDocument(key, value));
+  }
+
+  template <typename T>
+  ByteString Encode(T&& nanopb_proto) {
+    return MakeByteString(MakeMessage(std::move(nanopb_proto)));
```

Comment:
Given the argument is a forwarding reference, shouldn't this be `std::forward<T>`?